tests/components/alarm_control_panel/test_homekit_controller.py

```diff
@@ -0,0 +1,142 @@
+"""Basic checks for HomeKitAlarmControlPanel."""
+from unittest import mock
+
+from homeassistant.const import STATE_ALARM_ARMED_AWAY, ATTR_BATTERY_LEVEL
+from homeassistant.components.alarm_control_panel.homekit_controller import (
+    HomeKitAlarmControlPanel)
+
+
+def mock_alarm_control_panel():
+    """Return a HomeKitAlarmControlPanel attached to a fake device."""
+    from homekit.model.characteristics import CharacteristicsTypes
+
+    accessory = mock.Mock()
+    accessory.model = "TestCo"
+
+    pdata = accessory.pairing.pairing_data = {}
+    pdata['accessories'] = [{
+        "aid": 1,
+        "services": [{
+            "iid": 1,
+            "characteristics": [
+                {
+                    "iid": 2,
+                    "type": CharacteristicsTypes.SECURITY_SYSTEM_STATE_CURRENT,
+                    "perms": ["pr"],
+                },
+                {
+                    "iid": 3,
+                    "type": CharacteristicsTypes.SECURITY_SYSTEM_STATE_TARGET,
+                    "perms": ["pr"],
+                },
+                {
+                    "iid": 4,
+                    "type": CharacteristicsTypes.BATTERY_LEVEL,
+                    "perms": ["pr"],
+                },
+            ]
+        }]
+    }]
+
+    devinfo = {
+        "aid": 1,
+        "iid": 1,
+        "serial": "0000",
+    }
+
+    return HomeKitAlarmControlPanel(accessory, devinfo)
+
+
+def test_get_characteristic_types():
+    """Test the entity uses 1 or more characteristics."""
+    entity = mock_alarm_control_panel()
```

Comment:
Please don't test the entity in isolation. Set up the homekit controller component, via `async_setup_component`, let it set up the correct platform, let that add an entity. Then use the core to test the entity, eg by calling a service and asserting the result on the library methods. If it's a sensor, assert the state of the entity in the core state machine.

Make sure to patch and mock appropriately to avoid I/O.

Reply:
I usually recommend the deconz test suite.

We usually put common things in common.py under the component in question in the tests directory.

When using pytest fixtures, common fixtures go in conftest.py under the component in question. Pytest fixtures are awesome to use when writing tests that need common functionality.

Consider moving the platforms under the homekit controller component package. That's our new way of structuring components with platforms.

Reply:
Plan! One last question - is there a platform I can look at that does this already?

Reply:
Deconz. :smile:

Reply:
Great. So i'm trying to figure out how HA 'finds' light.py/cover.py in this new structure. Does `setup_platform` still work? Deconz, hue, etc use `setup_entry` instead.

Reply:
In my testing it looks like just `git mv homeassistant/components/light/homekit_controller.py homeassistant/components/homekit_controller/light.py` is enough!

Reply:
Magic! That logic is in loader.py.